### PR TITLE
Update versions and adapt instructions for trial

### PR DESCRIPTION
**Pull request by beckermarc**
Updated CDS Services versions to 1.2.0 and performed necessary code changes.
Updated instructions to explain how to use CloudPlatform Trial accounts.
Therefore also removed reference to XXX or other instructor provided details.

### CODE REVIEW

**Comment by beckermarc** [comment]
@agoerler Can you also have a look please :)

File: exercises-java/exercise1/README.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ Before you can start using SAP Application Studio, you need to create your devel
 
     ```bash
     mvn -B archetype:generate -DarchetypeArtifactId=cds-services-archetype -DarchetypeGroupId=com.sap.cds \
-    -DarchetypeVersion=1.0.1 -DcdsVersion=3.17.4 \
+    -DarchetypeVersion=1.2.0 -DcdsVersion=3.21.2 \
     -DgroupId=com.sap.teched.cap -DartifactId=products-service -Dpackage=com.sap.teched.cap.productsservice
```
Comment (chgeo) [inline exercises-java/exercise1/README.md:50]: We should replace this with a shorter call to `cds init`.  I guess, you prefer to still have the fixed archetype and cds versions in it, right?

Comment (beckermarc) [reply exercises-java/exercise1/README.md:50]: Yes, but the other problem is that we want to use version 1.2.0, but the CDS-DK version available on AppStudio only provides 1.1.0. Also we don't want to initialize sample data, which was surprisingly added by cds init by default.

Comment (chgeo) [reply exercises-java/exercise1/README.md:50]: So this means if we add the 3 parameters `archetypeVersion`, `cdsVersion`, and `samples` (or so) to cds init, it would work?
Would be something like
```
cds init products-service --add java --java:archetypeVersion=1.2.0 --java:cdsVersion=3.21.2
```
`samples` should default to `false` anyways, to align it with Node.js

Comment (beckermarc) [reply exercises-java/exercise1/README.md:50]: Yes, that would be great :)

Comment (chgeo) [reply exercises-java/exercise1/README.md:50]: Ok, I will align this with Joerg once I know his github user.

For the time being, we can leave the archetype call here.

File: exercises-java/exercise4/README.md [hunk 2]
```diff
@@ -85,7 +85,9 @@ You will now add a method to the `OrdersService` Java class to decrease the stoc
         @Before(event = CdsService.EVENT_CREATE, entity = "OrdersService.Orders")
         public void validateBookAndDecreaseStockViaOrders(List<Orders> orders) {
             for(Orders order : orders) {
-                validateBookAndDecreaseStock(order.getItems());
+                if(order.getItems() != null) {
```
Comment (agoerler) [inline exercises-java/exercise4/README.md:88]: why this change?

File: exercises-java/exercise4/README.md [hunk 3]
```diff
@@ -247,7 +249,9 @@ Finally, add a method to the `OrdersService` Java class to calculate the `total`
         public void calculateTotal(List<Orders> orders) {
             for (Orders order : orders) {
                 // calculate net amount for expanded items
-                calculateNetAmount(order.getItems());
+                if(order.getItems() != null) {
```
Comment (agoerler) [inline exercises-java/exercise4/README.md:252]: why this change?

**Review by agoerler — APPROVED** [review]

File: exercises-java/exercise4/README.md (hunk 2 above)
Comment (beckermarc) [reply exercises-java/exercise4/README.md:88]: CDS4J does not automatically initialize collection-based types with empty lists anymore upon calling the getter, but returns `null` now.
